src/groovy/heim/session/SessionCollector.groovy

```diff
@@ -0,0 +1,94 @@
+package heim.session
+
+import groovy.util.logging.Log4j
+import heim.tasks.JobTasksService
+import heim.tasks.TaskState
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Created by piotrzakrzewski on 22/10/15.
+ */
+@Log4j
+@Component
+class SessionCollector {
+
+    def SessionCollector(){
+        new Timer().schedule({
+            garbageCollection()
+        } as TimerTask, collectionInterval, collectionInterval)
+    }
+
+    @Autowired
+    private SessionService sessionService
```

Comment:
You would update the last modification time when a task *finished*, not when it's added/queued.